src/parsers/manifestjson.js

```diff
@@ -513,7 +513,18 @@ export default class ManifestJSONParser extends JSONParser {
         );
       }
       if (this.parsedJSON.background.service_worker) {
-        if (this.parsedJSON.manifest_version >= 3) {
+        if (!this.schemaValidatorOptions?.mv3BackgroundServiceWorker) {
+          // Report an error and mark the manifest as invalid if
+          // background service worker support isn't enabled by
+          // the addons-linter feature flags.
+          this.collector.addError(
+            messages.manifestFieldUnsupported(
+              '/background',
+              {} // mock schema error object.
```

Comment:
can you describe why we are doing this?

Reply:
it didn't seem worth adding a new specific linting message and so I used the pre-existing `manifestFieldUnsupported` one, which usually expects the ajv error to be passed as the second parameter, alternatively I could just tweak `manifestFieldUnsupported` implementation to account for the error object to not be always passed.

Reply:
> to account for the error object to not be always passed.

ha, if we can turn that parameter into an optional one, that'd be nice yeah.